pkg/auth/auth_test.go

```diff
@@ -221,3 +225,227 @@ func TestRevokeToken(t *testing.T) {
 		suite.Run(t, tsc)
 	}
 }
+
+// MockAPI is a mock implementation of the API interface
+type MockAPI struct {
+	mock.Mock
+}
+
+func (m *MockAPI) Login(ctx context.Context, req *auth.LoginRequest) (*auth.LoginResponse, error) {
+	args := m.Called(ctx, req)
+	return args.Get(0).(*auth.LoginResponse), args.Error(1)
+}
+
+func (m *MockAPI) RefreshToken(ctx context.Context, req *auth.RefreshTokenRequest) (*auth.RefreshTokenResponse, error) {
+	args := m.Called(ctx, req)
+	return args.Get(0).(*auth.RefreshTokenResponse), args.Error(1)
+}
+
+func (m *MockAPI) RevokeToken(ctx context.Context, req *auth.RevokeTokenRequest) error {
+	args := m.Called(ctx, req)
+	return args.Error(0)
+}
+
+type HelperClearTestSuite struct {
+	suite.Suite
+	mockAPI *MockAPI
+	helper  *auth.Helper
+}
+
+func (s *HelperClearTestSuite) SetupTest() {
+	s.mockAPI = new(MockAPI)
+	s.helper = &auth.Helper{
+		Username: "test_user",
+		Password: "test_password",
+		API:      s.mockAPI,
+	}
+	os.Remove("tokenstore.json")
+}
+
+func (suite *HelperClearTestSuite) TearDownTest() {
+	os.Remove("tokenstore.json")
+}
+
+func (s *HelperClearTestSuite) TestGetToken_NewToken() {
+	s.mockAPI.On("Login", mock.Anything, &auth.LoginRequest{Username: "test_user", Password: "test_password"}).Return(&auth.LoginResponse{
+		AccessToken:  "new_access_token",
+		RefreshToken: "new_refresh_token",
+	}, nil)
+
+	token, err := s.helper.GetAccessToken()
+
+	s.NoError(err)
+	s.Equal("new_access_token", token)
+	s.mockAPI.AssertExpectations(s.T())
+}
+
+func (s *HelperClearTestSuite) TestGetToken_ExistingValidToken() {
+	tokenStore := &auth.Tokenstore{
+		AccessToken:             "existing_access_token",
+		AccessTokenGeneratedAt:  time.Now(),
+		RefreshToken:            "existing_refresh_token",
+		RefreshTokenGeneratedAt: time.Now(),
+	}
+
+	data, _ := json.Marshal(tokenStore)
+	err := os.WriteFile("tokenstore.json", data, 0600)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	token, err := s.helper.GetAccessToken()
+
+	s.NoError(err)
+	s.Equal("existing_access_token", token)
+	s.mockAPI.AssertNotCalled(s.T(), "Login")
+	s.mockAPI.AssertNotCalled(s.T(), "RefreshToken")
+}
+
+func (s *HelperClearTestSuite) TestGetToken_ExpiredAccessToken() {
```

Comment:
As a convention, in Go, we do not use underscores for data structures, method names.

Rename other methods in this file as well.

Reply:
Renamed all tests